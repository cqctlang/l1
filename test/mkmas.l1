
printf("allocating buffer\n");
foo=malloc(1024);

printf("mutating malloc\n");
*(unsigned char *)foo = 1;

printf("mkmasing\n");
bar=mkmas(foo,1024);
printf("as: %a\n",bar);

printf("mutating mkmas\n");
*(unsigned char *){bar}foo = 1;

printf("mkmasroing\n");
baz=mkmasro(foo,1024);
printf("as: %a\n",baz);

printf("mutating mkmasro (should fail)\n");
*(unsigned char *){baz}foo = 1;
