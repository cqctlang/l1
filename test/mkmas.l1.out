allocating buffer
mutating malloc
mkmasing
as: <as>
mutating mkmas
mkmasroing
as: <as>
mutating mkmasro (should fail)
error: memory access fault
	#0                   roput	(builtin function)
	#1                   entry	(mkmas.l1:20)
